### ISSUE TITLE
migrate kubectl reference

### ISSUE THREAD

**Issue by kbhawkey** [kubernetes-sigs/reference-docs#118]
This is an umbrella issue to capture ideas for updating the kubectl command reference.
- Related to PR #83.
- The kubectl reference is difficult to read and navigate and is not easily maintained.
- The kubectl reference generation is error prone and typically the generated output has encountered a number of text formatting issues.

Come up with a solution to stabilize the kubectl command reference generation, UI, and code maintenance. Coordinate changes with the API reference generation (share assets, similar UI, similar generation process). As an alternative, the kubectl command reference could generate as a group of pages (Markdown) similar to other k8s/website reference tool pages. Is it important to readers to maintain a `reference` for the kubectl commands?

**Comment by tengqm** [comment]
One of the complaints we got is about the "examples". I'm not sure how useful they are but I'm unhappy about the fact that the examples layout is eating too much area of the screen.
For API reference, generating HTML directly is not optimal. There is still a tradeoff between having one huge HTML and having separate HTMLs for each group/kind/operation.
Generating markdown files from these generators and then generating HTMLs from these markdowns ... this doesn't make any senses to me. We may want to generate HTMLs (and other output formats) directly.
Consolidating all generators (API ref, kubectl ref, kube-* components...) to a single framework would be desirable.
We may need a redesign the framework so that:

- the same process can be used across all references;
- the look & feel of the generated (HTML) is consistent for all docs;
- the navigation across sections/subsections is made easier;
- the generation process is automated for users, especially the release team who may be not so familiar with the generators;
- the generation is highly customizable with configuration knobs exposed where appropriate.
- ...

**Comment by kbhawkey** [comment]
I agree with you about directly generating HTML (not Markdown). Some more ideas:
- The kubectl examples could be added to the main page for the command instead of in a separate container.
- The references could use a single code base or part of a common code base -- same backend and similar look-and-feel. Improve the navigation.

**Comment by kbhawkey** [comment]
I am temporarily assigning myself.
I started to update the code to directly generate HTML as well as clean up the css, js, toc.yaml. Will post what I have shortly.
/assign

**Comment by tengqm** [comment]
xref: kubernetes/website#14904

**Comment by tengqm** [comment]
xref: https://github.com/Birdrock/brodocs/pull/9 and the issue: https://github.com/kubernetes/website/issues/19519
